webapp/common/celery/celery.py

```diff
@@ -18,47 +18,106 @@
 from flask import Flask
-from flask_celery import Celery
 from kombu.serialization import register
 
 from webapp.common.json import DefaultJSONEncoder
 
 
+class CeleryState:
+    """Remembers the configuration for the (celery, app) tuple. Modeled from SQLAlchemy."""
+
+    celery: Celery
+    app: Flask
+
+    def __init__(self, celery: Celery, app: Flask):
+        self.celery = celery
+        self.app = app
+
+
 class LogErrorsCelery(Celery):
+    """Celery extension for Flask applications.
+
+    Involves a hack to allow views and tests importing the celery instance from extensions.py to access the regular
+    Celery instance methods. This is done by subclassing celery.Celery and overwriting celery._state._register_app()
+    with a lambda/function that does nothing at all.
+
+    That way, on the first super() in this class' __init__(), all of the required instance objects are initialized, but
+    the Celery application is not registered. This class will be initialized in extensions.py but at that moment the
+    Flask application is not yet available.
+
+    Then, once the Flask application is available, this class' init_app() method will be called, with the Flask
+    application as an argument. init_app() will again call celery.Celery.__init__() but this time with the
+    celery._state._register_app() restored to its original functionality. in init_app() the actual Celery application is
+    initialized like normal.
+    """
+
+    original_register_app: Callable
+
+    def __init__(self):
+        """If app argument provided then initialize celery using application config values.
+
+        If no app argument provided you should do initialization later with init_app method.
+
+        Keyword arguments:
+        app -- Flask application instance.
+        """
+        self.original_register_app = _state._register_app  # Backup Celery app registration function.
+        _state._register_app = lambda _: None  # Upon Celery app registration attempt, do nothing.
+        super().__init__()
+
     def init_app(self, app: Flask):
-        # register custom encoder, so Enumerations do not make json.dump exceptions In some tutorials they recommend
-        # registering an own ContentType when setting encoders because if you set content_type = application/json you
-        # will overwrite the old encoder. But if we just want another encoder, this is exactly what we want. We
-        # tested this, but if something breaks some time later this might be a place to start.
+        # register custom encoder, so Enumerations do not make json.dump exceptions
+        # In some tutorials they recommend registering an own ContentType when setting encoders because if you set
+        # content_type = application/json you will overwrite the old encoder. But if we just want another encoder, this
+        # is exactly what we want. We tested this, but if something breaks some time later this might be a place to
+        # start.
```

Comment:
```suggestion
                """ Register custom encoder, so Enumerations do not make json.dump exceptions
        In some tutorials they recommend registering an own ContentType when setting encoders because if you set
        content_type = application/json you will overwrite the old encoder. But if we just want another encoder, this
        is exactly what we want. We tested this, but if something breaks some time later this might be a place to
        start."""
```

Reason for my suggestion: usually we use `""" bla """` to write pydoc, but here we have `# bla` instead, which is not consistent. Plus the first letter of the first sentence is lowercase, which is also inconsistent.

Reply:
Done.